src/main/java/com/idrsolutions/microservice/JPedalServlet.java

```diff
@@ -270,88 +273,99 @@ protected boolean validateRequest(final HttpServletRequest request, final HttpSe
         switch (mode) {
             case convertToImages:
-                ConvertPagesToImages.writeAllPagesAsImagesToDir(
-                        userPdfFilePath,
-                        outputDirStr + fileSeparator + fileNameWithoutExt + fileSeparator,
-                        paramMap.get("format"),
-                        Float.parseFloat(paramMap.getOrDefault("scaling", "1.0")),
-                        paramMap.get("password"),
-                        durationTracker);
+                commandArgs.add("org.jpedal.examples.images.ConvertPagesToImages");
+                commandArgs.add(inputPdf.getAbsolutePath());
+                commandArgs.add(outputDir.getAbsolutePath());
+                commandArgs.add(conversionParams.get("format"));
+                commandArgs.add(String.valueOf(scaling));
                 break;
-                case extractImages:{
-                final String type = paramMap.get("type");
+            case extractImages:
+                String type = conversionParams.get("type");
                 switch (type) {
                     case "rawImages" :
-                        ExtractImages.writeAllImagesToDir(
-                                userPdfFilePath,
-                                paramMap.get("password"),
-                                outputDirStr + fileSeparator + fileNameWithoutExt + fileSeparator ,
-                                paramMap.get("format"), true, true,
-                                durationTracker);
+                        commandArgs.add("org.jpedal.examples.images.ExtractImages");
+                        commandArgs.add(inputPdf.getAbsolutePath());
+                        commandArgs.add(outputDir.getAbsolutePath());
+                        commandArgs.add(conversionParams.get("format"));
+
                         break;
                     case "clippedImages" :
-                        ExtractClippedImages.writeAllClippedImagesToDirs(userPdfFilePath,
-                                paramMap.get("password"),
-                                outputDirStr + fileSeparator,
-                                paramMap.get("format"),new String[]{"0",fileNameWithoutExt},
-                                durationTracker);
+                        String name = inputPdf.getName();
+                        commandArgs.add("org.jpedal.examples.images.ExtractClippedImages");
+                        commandArgs.add(inputPdf.getAbsolutePath());
+                        commandArgs.add(outputDir.getAbsolutePath());
+                        commandArgs.add(conversionParams.get("format"));
+                        commandArgs.add("0");
+                        commandArgs.add(name.substring(0, name.lastIndexOf(".")));
                         break;
                 }
-                } break;
+                break;
             case extractText:
-                final String type = paramMap.get("type");
+                type = conversionParams.get("type");
```

Comment:
Thoughts on renaming the variable so it can be final?
```suggestion
                final textType = conversionParams.get("type");
```